pootle/static/js/editor/index.js

```diff
@@ -12,23 +12,25 @@ import React from 'react';
 import ReactRenderer from 'utils/ReactRenderer';
 import { q, qAll } from 'utils/dom';
 
-import Editor from './containers/Editor';
-import UnitSource from './components/UnitSource';
+import { loadFormatAdaptor } from './formats/FormatLoader';
 import { hasCRLF, normalize, denormalize } from './utils/normalizer';
 
 
 const ReactEditor = {
 
   init(props) {
-    this.node = q('.js-mount-editor');
-    this.sourceNode = q('.js-mount-editor-original-src');
-    this.alternativeSourceNodes = qAll('.js-mount-editor-alt-src');
-    this.props = {};
-    this.hasCRLF = props.sourceValues.some(hasCRLF);
-
-    ReactRenderer.unmountComponents();
-
-    this.setProps(props);
+    loadFormatAdaptor(props, (newProps, adaptor) => {
```

Comment:
I see the intention to pass more data than file type and source values although I'm still inclined to passing arguments explicitly. We never know how can the init'ing code can grow in its arguments and we might end up passing a lot of useless properties.